src/redux/preview.js

```diff
@@ -95,6 +95,38 @@ module.exports.previewReducer = (state, action) => {
                 return comment;
             })
         });
+    case 'ADD_NEW_COMMENT':
+        if (action.comment.parent_id) {
+            let topLevelParent = action.comment.parent_id;
+
+            // If this is a nested reply, we need to look up which top level comment
+            // to put this new reply under.
```

Comment:
Why not look up the parent's parent't parent's... parent instead? If we haven't introduced errors, that should work, right? And be faster (though I'm not sure how much it really matters since we only do all this once per added comment)